.github/workflows/release.yml

```diff
@@ -13,37 +13,55 @@ env:
   HUSKY: 0
 
 jobs:
-  release:
-    concurrency: ${{ github.workflow }}-${{ github.ref }}
+  get-branch-name:
+    runs-on: ubuntu-latest
+    outputs:
+      current_branch: ${{ steps.branch-name.outputs.current_branch }}
+    steps:
+      - name: Get branch name
+        id: branch-name
+        uses: tj-actions/branch-names@v7.0.7
+  check-dialpad-member:
     runs-on: ubuntu-latest
-
-    strategy:
-      matrix:
-        node-version: [ 20 ]
-
     steps:
-      - uses: actions/checkout@v3
-      - uses: pnpm/action-setup@v2
+      # Will prevent the rest of the steps from running on fail
+      - name: Check if user is a dialpad member
+        uses: octokit/request-action@v2.x
         with:
-          version: 8
+          route: GET /orgs/dialpad/members/${{ github.actor }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.DIALTONE_CI_TOKEN }}
+  npm:
+    needs: [get-branch-name, check-dialpad-member]
+    concurrency: ${{ github.workflow }}-${{ github.ref }}-npm
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
 
-      - name: Use Node.js ${{ matrix.node-version }}
+      - name: Use Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node-version }}
-          cache: 'pnpm'
+          node-version: 20
 
-      - name: parse branch from ref
-        id: branch
-        run: echo ::set-output name=BRANCH::${GITHUB_REF/refs\/heads\//}
+      - name: Use pnpm
+        uses: pnpm/action-setup@v2
+        with:
+          version: 8
+          run_install: false
 
-      # Will prevent the rest of the steps from running on fail
-      - name: Check if user is a dialpad member
-        uses: octokit/request-action@v2.1.0
+      - name: Get pnpm store directory
+        shell: bash
+        run: |
+          echo "STORE_PATH=$(pnpm store path --silent)" >> $GITHUB_ENV
+
+      - name: Setup pnpm cache
+        uses: actions/cache@v3
         with:
-          route: GET /orgs/dialpad/members/${{ github.actor }}
-        env:
-          GITHUB_TOKEN: ${{ secrets.DIALTONE_CI_TOKEN }}
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
 
       - name: Install dependencies
         run: pnpm install --frozen-lockfile
```

Comment:
Do we want to skip the install if we got a cache hit? or it doesn't matter that much?

Reply:
Doesn't matter much, without cache pnpm takes like 2 minutes and with cache 8 seconds. 
That's the recommended implementation from the pnpm documentation.

Reply:
k sounds good.